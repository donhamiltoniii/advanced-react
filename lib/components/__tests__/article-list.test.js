import React from 'react';
import ArticleList from '../article-list';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {
  const testProps = {
    actions: {
      lookupAuthor: jest.fn(() => ({}))
    },
    articles: {
      a: { id: 'a' },
      b: { id: 'b' }
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();

    expect(tree.children.length).toEqual(2);
    expect(tree).toMatchSnapshot();
  });
});
