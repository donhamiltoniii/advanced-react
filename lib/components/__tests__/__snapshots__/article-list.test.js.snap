// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div
  className="article-list"
>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "color": "white",
      }
    }
  >
    <div />
    <div />
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
  <div
    style={
      Object {
        "backgroundColor": "red",
        "color": "white",
      }
    }
  >
    <div />
    <div />
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
